src/routes/safe/components/Transactions/TxsTable/index.tsx

```diff
@@ -28,40 +28,50 @@ const TxsTable = ({ classes }) => {
   const [expandedTx, setExpandedTx] = useState(null)
   const cancellationTransactions = useSelector(safeCancellationTransactionsSelector)
   const transactions = useSelector(extendedTransactionsSelector)
+  const [filteredData, setFilteredData] = useState(null)
 
   const handleTxExpand = (safeTxHash) => {
     setExpandedTx((prevTx) => (prevTx === safeTxHash ? null : safeTxHash))
   }
 
   const columns = generateColumns()
   const autoColumns = columns.filter((c) => !c.custom)
-  const filteredData = getTxTableData(transactions, cancellationTransactions)
-    .sort((tx1, tx2) => {
-      // First order by nonce
-      const aNonce = tx1.tx.nonce
-      const bNonce = tx1.tx.nonce
-      if (aNonce && bNonce) {
-        const difference = aNonce - bNonce
-        if (difference !== 0) {
-          return difference
-        }
-      }
-      // If can't be ordered by nonce, order by date
-      const aDateOrder = tx1.dateOrder
-      const bDateOrder = tx2.dateOrder
-      // Second by date
-      if (!aDateOrder || !bDateOrder) {
-        return 0
-      }
-      return aDateOrder - bDateOrder
-    })
-    .map((tx, id) => {
-      return {
-        ...tx,
-        id,
-      }
-    })
 
+  useMemo(() => {
```

Comment:
I recently learned about `useMemo`. It `returns` a _memoized_ value: https://reactjs.org/docs/hooks-reference.html#usememo

So, instead of setting state values from within a `useMemo`, you should return that value.

BTW: _Don’t do anything there that you wouldn’t normally do while rendering. For example, side effects belong in useEffect, not useMemo._